app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'post_id', class_name: 'Like'
+  has_many :comments, foreign_key: 'post_id', class_name: 'Comment'
+
+  belongs_to :author, class_name: 'User'
+
+  def update_post_counter
+    user.increment!(:post_counter)
+  end
```

Comment:
- [ ] Since a post belongs to an `author` and not a `user`, kindly change this method to use `author` instead.

  ```
  def update_post_counter
    author.increment!(:post_counter)
  end
  ```

- [ ] **[OPTIONAL]** Instead of having to call this method yourself when creating a post, I would recommend you implement the `after_save` active record callback so that the author's post counter is automatically incremented after a post is saved.

  ```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:post_counter)
  end
  ```